src/app/qgisapp.cpp

```diff
@@ -10001,7 +10001,23 @@ void QgisApp::reshapeFeatures()
 
 void QgisApp::addFeature()
 {
-  mMapCanvas->setMapTool( mMapTools->mapTool( QgsAppMapTools::AddFeature ) );
+  if ( mMapCanvas->mapTool()->toolName() != tr( "Add feature" ) )
```

Comment:
```suggestion
  if ( mMapCanvas->mapTool() != mMapTools->mapTool( QgsAppMapTools::AddFeature ) )
```

That's safer then relying on a translated string.